### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by Mahdi-Niazi**
### In this milestone I:

- Add user model
- Add post model
- Add like model
- Add comments model
- Add recent_post method
- Add update_post_counter
- Add recent_comments
- Add update_like_counter method
- Add update_comment_counter method

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    User.find(author_id).update(post_counter: Post.where(author_id).count)
+  end
```
Comment (Qoosim) [inline app/models/post.rb:12]: - [ ] Great job for making use of one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), `after_save`. it is also good to make this method `private`. Kindly make it private

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_save :update_comment_counter
+
+  def update_coumment_counter
+    Post.find(post_id).update(comments_counter: Comment.where(post_id).count)
+  end
```
Comment (Qoosim) [inline app/models/comment.rb:11]: - [ ] Great job for making use of one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), `after_save`. it is also good to make this method `private`. Kindly make it private. It should look something similar to this :point_down: 

```
  class Comment < ApplicationRecord
  belongs_to :author, class_name: 'User'
  belongs_to :post

  after_save :update_comment_counter

  private
  def update_coumment_counter
    Post.find(post_id).update(comments_counter: Comment.where(post_id).count)
  end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_save :update_like_counter
+
+  def update_like_counter
+    Post.find(post_id).update(likes_counter: Like.where(post_id).count)
+  end
```
Comment (Qoosim) [inline app/models/like.rb:11]: - [ ] Great job for making use of one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), `after_save`. it is also good to make this method `private`. Kindly make it private. It should look something similar to this :point_down: 

```
  class Like < ApplicationRecord
  belongs_to :author, class_name: 'User'
  belongs_to :post

  after_save :update_like_counter

private  
def update_like_counter
    Post.find(post_id).update(likes_counter: Like.where(post_id).count)
  end
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Mahdi-Niazi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @Mahdi-Niazi  👍 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Rails features  ✔️  
- [x] Kudus on meeting the requirements   👏 


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**